parser/src/lib.rs

```diff
@@ -204,22 +204,29 @@ fn parse_document_blocks(input: &str) -> IResult<&str, Vec<Ast>> {
 ///
 /// returns: The heading node, if a successful parse occurs, otherwise the parse error
 fn parse_heading(input: &str) -> IResult<&str, Heading> {
-    map(
+    let res = map(
         pair(
             verify(take_while1(|c| c == '#'), |s: &str| {
                 s.len() <= u8::MAX as usize
             }),
             preceded(space0, parse_heading_text),
         ),
-        |(start, body)| {
-            let mut elements = tag::extract_tags(vec![Text(body.into())]);
```

Comment:
Isn't it enough just to replace the `tag::extract_tags` and the `smart_punctuate` with `parse_paragraph_elements`, and that's everything we need to do? Right? (And change map to map_parser)

Reply:
Good idea! I didn't know ``map_parser`` worked like that. Now that I'm reading about it, it seems like it may even be possible to just use ``.and_then(parse_paragraph_elements)`` on top of ``preceded``.